### PR TITLE
Implement zoomToFitMarkers

### PR DESCRIPTION
**Pull request by jvartanian94**
Add the ability to zoom map to fit a given set of AirMapMarkers

### CODE REVIEW

File: library/src/main/java/com/airbnb/android/airmapview/NativeGoogleMapFragment.java [hunk 1]
```diff
@@ -231,6 +232,23 @@ public void onMapClick(LatLng latLng) {
     polyline.removeFromGoogleMap();
   }
 
+  @Override
+  public void zoomToFitMarkers(int padding, AirMapMarker... markers) {
+    LatLngBounds.Builder b = new LatLngBounds.Builder();
```
Comment (petzel) [inline library/src/main/java/com/airbnb/android/airmapview/NativeGoogleMapFragment.java]: rename this to `latLngBounds`

File: library/src/main/java/com/airbnb/android/airmapview/AirMapInterface.java [hunk 2]
```diff
@@ -188,4 +188,10 @@
    * @param polyline the {@link Polyline} to remove
    */
   void removePolyline(AirMapPolyline polyline);
+
+  /**
+   * Zoom map to fit the given {@Link AirMapMarker} markers
+   */
+
+  void zoomToFitMarkers(int padding, AirMapMarker... markers);
```
Comment (petzel) [inline library/src/main/java/com/airbnb/android/airmapview/AirMapInterface.java]: can we change this to take in `List<AirMapMarker> markers` instead of varargs

File: library/src/main/java/com/airbnb/android/airmapview/NativeGoogleMapFragment.java [hunk 3]
```diff
@@ -231,6 +232,23 @@ public void onMapClick(LatLng latLng) {
...
+    for(AirMapMarker marker : markers){
+      b.include(marker.getLatLng());
+    }
+    LatLngBounds bounds = b.build();
+    final CameraUpdate cu = CameraUpdateFactory
```
Comment (petzel) [inline library/src/main/java/com/airbnb/android/airmapview/NativeGoogleMapFragment.java]: `cameraUpdate` instead of cu

File: library/src/main/java/com/airbnb/android/airmapview/NativeGoogleMapFragment.java [hunk 4]
```diff
@@ -231,6 +232,23 @@ public void onMapClick(LatLng latLng) {
...
+            .newLatLngBounds(bounds, padding);
+    googleMap.setOnMapLoadedCallback(new GoogleMap.OnMapLoadedCallback() {
```
Comment (petzel) [inline library/src/main/java/com/airbnb/android/airmapview/NativeGoogleMapFragment.java]: if the map is already loaded, this is not necessary to add this loaded callback. also, this will overwrite any map loaded callback that may have been set previously

**Comment by petzel** [comment]
This is great! A few comments to address before we can merge this

File: library/src/main/java/com/airbnb/android/airmapview/NativeGoogleMapFragment.java (hunk 4 above)
Comment (jvartanian94) [inline library/src/main/java/com/airbnb/android/airmapview/NativeGoogleMapFragment.java]: @petzel I need to figure out a different way to set this callback without overwriting any existing callbacks then because zoomToFitMarkers can be called after onMapInitialized but the MapView isn't inflated yet. This results in a force close with the following error:

"java.lang.IllegalStateException: Error using newLatLngBounds(LatLngBounds, int): Map size can't be 0"

**Comment by jvartanian94** [comment]
onMapInitialized is called before onMapLoaded. In order to fit the native map to the given markers, it must be completely loaded first

File: library/src/main/java/com/airbnb/android/airmapview/AirMapInterface.java [hunk 5]
```diff
@@ -128,6 +131,8 @@
 
   void setOnMapLoadedListener(OnMapLoadedListener onMapLoadedListener);
 
+  void setOnMapInitialisedListener(OnMapInitializedListener onMapInitialisedListener);
```
Comment (petzel) [inline library/src/main/java/com/airbnb/android/airmapview/AirMapInterface.java]: type Initialized vs Initialised

File: library/src/main/java/com/airbnb/android/airmapview/WebViewMapFragment.java [hunk 6]
```diff
@@ -254,6 +281,9 @@ public void run() {
             if (onMapLoadedListener != null) {
               onMapLoadedListener.onMapLoaded();
             }
+            if (onMapInitializedListener != null) {
```
Comment (petzel) [inline library/src/main/java/com/airbnb/android/airmapview/WebViewMapFragment.java]: we should call `onMapInitialized` before `onMapLoaded` as that's how the native map will behave

File: library/src/main/java/com/airbnb/android/airmapview/WebViewMapFragment.java [hunk 7]
```diff
@@ -156,6 +158,15 @@ public void setOnCameraChangeListener(OnCameraChangeListener listener) {
     onCameraChangeListener = listener;
   }
 
+  @Override
+  public void setOnMapInitialisedListener(OnMapInitializedListener listener) {
+    this.onMapInitializedListener = listener;
+    if (loaded) {
+      onMapInitializedListener.onMapInitialized();
```
Comment (petzel) [inline library/src/main/java/com/airbnb/android/airmapview/WebViewMapFragment.java]: why is this needed? it's not actually initialized at this point -- if someone sets a new initialized listener far after it's initialized, we don't want to re-run any initialization code the developer may have added. we should remove this

File: library/src/main/java/com/airbnb/android/airmapview/WebViewMapFragment.java [hunk 8]
```diff
@@ -156,6 +158,15 @@ public void setOnCameraChangeListener(OnCameraChangeListener listener) {
     onCameraChangeListener = listener;
   }
 
+  @Override
+  public void setOnMapInitialisedListener(OnMapInitializedListener listener) {
```
Comment (petzel) [inline library/src/main/java/com/airbnb/android/airmapview/WebViewMapFragment.java]: typo `Initialized`

File: library/src/main/java/com/airbnb/android/airmapview/WebViewMapFragment.java (hunk 7 above)
Comment (jvartanian94) [inline library/src/main/java/com/airbnb/android/airmapview/WebViewMapFragment.java]: @petzel that's how you guys had it in the original setOnMapLoadedListener. I will remove it
